lib/java-extras/src/main/java/org/triplea/java/Retryable.java

```diff
@@ -3,85 +3,82 @@
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
 import java.time.Duration;
-import java.util.Optional;
+import java.util.function.BooleanSupplier;
 import java.util.function.Consumer;
-import java.util.function.Supplier;
 import lombok.AccessLevel;
 import lombok.AllArgsConstructor;
 
 /**
  * Module to execute a task with retries. Provides a builder interface to specify number of max
  * attempts (max number of times the task will be executed) and backoff.
- *
- * @param <T> Generic type that is returned by the retryable.
  */
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
-public class Retryable<T> {
+public class Retryable {
   private static final Consumer<Duration> DEFAULT_THREAD_SLEEP =
       duration -> Interruptibles.sleep(duration.toMillis());
 
   private final Consumer<Duration> threadSleeper;
   private final int maxAttempts;
   private final Duration fixedBackOff;
-  private final Supplier<Optional<T>> taskRunner;
+  private final BooleanSupplier taskRunner;
```

Comment:
Is this update to take into account that `T` is so far only a boolean?

This `Retryable` class was meant to be generic, but I don't think it turned out that way very nicely. There might be a better solution to this class, namely deleting it or rewriting it so it is more usable as a generic retry mechanism.

Reply:
Redone in commit [94b829759c377f9411cf58033c9ae750e3443db8](https://github.com/triplea-game/triplea/pull/9939/commits/94b829759c377f9411cf58033c9ae750e3443db8)